### PR TITLE
I737 4 goal edit modal milestones

### PR DESCRIPTION
**Pull request by Dartv**

### CODE REVIEW

**Comment by Dartv** [comment]
Wow, this one is HUGE 😲

File: share/services/goal-service.js [hunk 1]
```diff
@@ -62,14 +63,43 @@ export default {
...
       _id,
       isDeleted: true,
       deletedBy: userId,
       deletedAt: new Date(),
     });
+
+    const { goal: { milestoneIds } } = res;
+
+    await Promise.all(milestoneIds.map(milestoneId =>
+      MilestoneService.delete({ _id: milestoneId }, { userId })));
```
Comment (Serhiy91) [inline share/services/goal-service.js]: Are you sure that we need to delete Milestones? We can just remove it when goal removing permanently

Comment (Dartv) [reply share/services/goal-service.js]: Why though? What if we need to filter only existing milestones when performing tasks in `background-app`?

Comment (Serhiy91) [reply share/services/goal-service.js]: Then need to restore Milestones when restoring Goal

**Pull request merged**